File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
# Q1710e runtime 39.68%
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x : x[1], reverse=True)
        ans = cntbox = 0
        for numbox, unit in boxTypes:
            if cntbox + numbox <= truckSize:
                ans += unit * numbox
                cntbox += numbox
            else:
                ans += unit * (truckSize - cntbox)
                break
        return ans
    
'''
WA
# forgot to return
[[1,3],[5,5],[2,5],[4,2],[4,1],[3,1],[2,2],[1,3],[2,5],[3,2]]
35
'''

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.go
func maximumUnits(boxTypes [][]int, truckSize int) int {
    sort.Slice(boxTypes, func(i, j int) bool {
        return boxTypes[i][1] > boxTypes[j][1]
    })
    var sumbox, sumunit int = 0, 0
    for _, box := range boxTypes {
        if sumbox + box[0] <= truckSize {
            sumunit += box[1] * box[0]
            sumbox += box[0]
        } else {
            sumunit += box[1] * (truckSize - sumbox)
            break
        }
    }
    return sumunit
}